docker/clientbase.py

```diff
@@ -221,6 +221,46 @@ def _multiplexed_response_stream_helper(self, response):
                 break
             yield data
 
+    def _stream_raw_result_old(self, response):
+        ''' Stream raw output for API versions below 1.6 '''
+        self._raise_for_status(response)
+        for line in response.iter_lines(chunk_size=1,
+                                        decode_unicode=True):
+            # filter out keep-alive new lines
+            if line:
+                yield line
+
+    def _stream_raw_result(self, response):
+        ''' Stream result for TTY-enabled container above API 1.6 '''
+        self._raise_for_status(response)
+        for out in response.iter_content(chunk_size=1, decode_unicode=True):
+            yield out
+
+    def _get_result(self, container, stream, res):
+        cont = self.inspect_container(container)
```

Comment:
As an optimisation, `attach` (and `_get_result`) could take an optional argument to force TTY or non-TTY behaviour. Syntax to be determined, but e.g.:

```python
client.attach(force_tty=True)
client.attach(force_no_tty=True)
```

The `inspect_container` call could then be skipped.

This wouldn't need to be implemented as part of this PR - the pressing concern is to fix the bug.